slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/OldPlainSQLTest.scala

```diff
@@ -110,15 +110,55 @@ class OldPlainSQLTest extends TestkitTest[JdbcTestDB] {
     println("User for ID 2: "+res)
     assertEquals(User(2,"guest"), res)
 
-    val s1 = sql"select id from USERS where name = ${"szeiger"}".as[Int]
-    val s2 = sql"select id from USERS where name = '#${"guest"}'".as[Int]
-    assertEquals("select id from USERS where name = ?", s1.getStatement)
+    val s1 = sql"select id from USERS where name = '${"szeiger"}'".as[Int]
+    val s2 = sql"select id from USERS where name = '${"guest"}'".as[Int]
+    assertEquals("select id from USERS where name = 'szeiger'", s1.getStatement)
```

Comment:
This is a very surprising semantic difference. The interpolators should not behave differently when you use a constant vs a non-constant parameter. Such a difference can be used for optimization but not for semantics.

We cannot leave out `#` either. It is essential for building queries dynamically (as in the build failure). Of course, this is not possible for `tsql`, so any use of a `#` parameter in tsql should cause compilation to fail, but it is an essential feature for `sql` and `sqlu`. A typical use case is splicing in a table name dynamically.